plugins/org.jkiss.dbeaver.ext.databricks/src/org/jkiss/dbeaver/ext/databricks/DatabricksDataSource.java

```diff
@@ -35,4 +37,18 @@ public DatabricksDataSource(DBRProgressMonitor monitor,
     }
 
 
+    @Override
+    protected String getConnectionURL(DBPConnectionConfiguration connectionInfo) {
+        String url = super.getConnectionURL(connectionInfo);
+        if (!isLegacyDriver() && url.startsWith(DatabricksConstants.JDBC_LEGACY_URL_SUBPROTOCOL)) {
```

Comment:
Is it expected here to have **!**` isLegacyDriver()`? Because logic is inverted in this case.